framework/resource/retryresource/basic_resource.go

```diff
@@ -0,0 +1,116 @@
+package retryresource
+
+import (
+	"context"
+	"fmt"
+	"time"
+
+	"github.com/cenkalti/backoff"
+	"github.com/giantswarm/microerror"
+	"github.com/giantswarm/micrologger"
+	"github.com/giantswarm/operatorkit/framework"
+)
+
+type basicResourceConfig struct {
+	Logger micrologger.Logger
+	//Resource framework.Resource
+	Resource interface {
+		Name() string
+		Underlying() framework.Resource
+
+		EnsureCreated(ctx context.Context, obj interface{}) error
+		EnsureDeleted(ctx context.Context, obj interface{}) error
+	}
+
+	BackOff backoff.BackOff
+}
+
+type basicResource struct {
+	logger micrologger.Logger
+	//resource framework.Resource
+	resource interface {
+		Name() string
+		Underlying() framework.Resource
+
+		EnsureCreated(ctx context.Context, obj interface{}) error
+		EnsureDeleted(ctx context.Context, obj interface{}) error
+	}
+
+	backOff backoff.BackOff
+}
+
+func newBasicResource(config basicResourceConfig) (*basicResource, error) {
```

Comment:
Why is this private? How do you want to use it?

Reply:
https://github.com/giantswarm/operatorkit/pull/153/files#diff-b5702ae7a6bf60d3a1767213f2a0f140R64